### DISCUSSION TITLE
Default configuration

### DISCUSSION THREAD

**Discussion by Toso82** [SixLabors/ImageSharp.Web#292] [Q&A] [answered]
I don't find it is possible to setting some default configuratio. I mean i want to setting for all web request without pass parameter:

1.  Default Background color
2.  Default format save image

**Comment by JimBobSquarePants** [comment, marked as answer]
Yeah. You can set them in [ImageSharpMiddlewareOptions.OnParseCommands](https://docs.sixlabors.com/api/ImageSharp.Web/SixLabors.ImageSharp.Web.Middleware.ImageSharpMiddlewareOptions.html#SixLabors_ImageSharp_Web_Middleware_ImageSharpMiddlewareOptions_OnParseCommandsAsync)

```c#
services.AddImageSharp(options =>
{
    options.OnParseCommandsAsync = context => {
        context.Commands.Add(nameof(BackgroundColorWebProcessor.Color), nameof(Color.White));
        context.Commands.Add(nameof(FormatWebProcessor.Format), "png");
        return Task.CompletedTask;
    };                
}) 
```

**Reply by Toso82** [reply]
Thanks good to know

**Comment by Toso82** [comment]
@JimBobSquarePants  I have another problem it is possible to change default url request to change image? I mean i want to add some prefix  img/{PATH_TO_YOUR_IMAGE}?width=300 for example. Actualy use Azure Storage

**Reply by JimBobSquarePants** [reply]
There’s an Azure Storage provider. See the docs I linked to before

**Reply by Toso82** [reply]
Yes but i don't find there is possibility to add some prefix on url of request 

/{CONTAINER_NAME}/{BLOB_FILENAME} 

I want something like

BLOB/{CONTAINER_NAME}/{BLOB_FILENAME}

**Reply by JimBobSquarePants** [reply]
Then implement your own provider. The whole thing is extensible

**Reply by Toso82** [reply]
I see but i must to do one similar with this options if there is not other possibility to use this could be interest to add something to add prefix when someone it has problem with url conflict

**Reply by JimBobSquarePants** [reply]
No. We offer common functionality and the opportunity for you to implement your own custom implementations.
